src/network/mod.rs

```diff
@@ -75,50 +87,128 @@ pub fn init(
 		id_keys.public()
 	);
 
-	// create transport
-	let transport = setup_transport(&id_keys, cfg.libp2p_tcp_port_reuse)?;
-
-	// create swarm that manages peers and events
-	let swarm = {
-		let mut kad_cfg = KademliaConfig::default();
-		kad_cfg
-			.set_publication_interval(cfg.kademlia.publication_interval)
-			.set_replication_interval(cfg.kademlia.record_replication_interval)
-			.set_replication_factor(cfg.kademlia.record_replication_factor)
-			.set_connection_idle_timeout(cfg.kademlia.connection_idle_timeout)
-			.set_query_timeout(cfg.kademlia.query_timeout)
-			.set_parallelism(cfg.kademlia.query_parallelism)
-			.set_caching(KademliaCaching::Enabled {
-				max_peers: cfg.kademlia.caching_max_peers,
-			})
-			.disjoint_query_paths(cfg.kademlia.disjoint_query_paths);
-
-		let store_cfg = MemoryStoreConfig {
+	let noise_keys = Keypair::<X25519Spec>::new().into_authentic(&id_keys)?;
+
+	// Create Transport
+	// relay transport configuration used in relay clients
+	let local_peer_id = PeerId::from(id_keys.public());
+	let (relay_client_transport, relay_client_behaviour) = relay::client::new(local_peer_id);
+
+	let mut yamux_config = YamuxConfig::default();
+	// enabling propper flow-control:
+	// window updates are only sent when buffered data has been consumed
```

Comment:
If the receiver is too slow, in our case LC is bogged down for whatever reason, this could potentially cause back-pressure to slow down the client that's sending the data e.g. the fat client. 

It might be better to set it to `on_receive()`. wdyt?

Reply:
that mode introduces a possibility for deadlock
we should discuss this further